test/lightning_web/end_to_end_test.exs

```diff
@@ -1,173 +1,278 @@
 # This module will be re-introduced in https://github.com/OpenFn/Lightning/issues/1143
 defmodule LightningWeb.EndToEndTest do
-  use LightningWeb.ConnCase, async: true
-  use Oban.Testing, repo: Lightning.Repo
+  use LightningWeb.ConnCase, async: false
 
   import Lightning.JobsFixtures
   import Lightning.Factories
 
+  alias Lightning.Attempt
+  alias Lightning.Attempts
+  alias Lightning.Attempts.Events
   alias Lightning.Invocation
+  alias Lightning.Repo
+  alias Lightning.WorkOrders
+  alias Lightning.Runtime.RuntimeManager
 
-  import Ecto.Query
+  require Attempt
 
-  setup :register_and_log_in_superuser
+  setup_all context do
+    start_runtime_manager(context)
+  end
+
+  describe "webhook triggered attempts" do
+    setup :register_and_log_in_superuser
+
+    @tag timeout: 120_000
+    test "complete an attempt on a complex workflow with parallel jobs", %{
+      conn: conn
+    } do
+      project = insert(:project)
+
+      %{triggers: [%{id: webhook_trigger_id}]} =
+        insert(:complex_workflow, project: project)
+
+      # Post to webhook
+      webhook_body = %{"x" => 1}
+      conn = post(conn, "/i/#{webhook_trigger_id}", webhook_body)
+
+      assert %{"work_order_id" => wo_id} = json_response(conn, 200)
+
+      assert %{attempts: [%{id: attempt_id}]} =
+               WorkOrders.get(wo_id, include: [:attempts])
+
+      assert %{runs: []} = Attempts.get(attempt_id, include: [:runs])
+
+      assert %{attempts: [attempt]} =
+               WorkOrders.get(wo_id, include: [:attempts])
+
+      # wait to complete
+      Events.subscribe(attempt)
+
+      Enum.any?(1..100, fn _i ->
+        receive do
+          %Events.AttemptUpdated{attempt: %{state: state}}
+          when state in Attempt.final_states() ->
+            assert state == :success
+            true
+
+          _other_msg ->
+            false
+        end
+      end)
```

Comment:
How many messages can this process receive? In the failure case (where the `:success` state doesn't arrive), I guess the number of iterations is not relevant but `receive` waits indefinitely by default - looks like we would have to wait the entire test timeout for this. 

:man_shrugging: I guess it's fine... Can you think of ways this could be improved? If not then it's probably fine :sweat_smile:

Reply:
There are multiple messages, any 100 is just a big number to make it simple since it's a test setup only. I could use a stream but this is simpler and still readable.